Integration tests fail (based on https://github.com/dmn-tck/tck)

__What should we do?__

When running the integration tests (based on https://github.com/dmn-tck/tck), the following test case fails:
```
  1) dmn-moddle - TCK roundtrip
       should roundtrip
         tmp/tck/TestCases/compliance-level-3/1130-feel-interval/1130-feel-interval.dmn:
     AssertionError: expected [ Array(1) ] to be empty
```

This can be reproduced using

```sh
$ npm i
$ npm run test:integration
```

The goal of this issue is to investigate what casues this and what this means for us.

__Context__

Tested with https://github.com/bpmn-io/dmn-moddle/commit/bb8782a09e100b846dcfdfea362d8e3d75c7ae45

--- a/resources/dmn/json/di.json
+++ b/resources/dmn/json/di.json
@@ -8,6 +8,10 @@
       "isAbstract": true,
       "properties": [
         {
+          "name": "extension",
+          "type": "Extension"
+        },
+        {
           "name": "id",
           "isAttr": true,
           "isId": true,
@@ -91,6 +95,16 @@
           "isAttr": true,
           "isId": true,
           "type": "String"
+        }
+      ]
+    },
+    {
+      "name": "Extension",
+      "properties": [
+        {
+          "name": "values",
+          "isMany": true,
+          "type": "Element"
         }
       ]
     }

--- a/tasks/transforms/transformDI.js
+++ b/tasks/transforms/transformDI.js
@@ -59,6 +59,10 @@ module.exports = async function(results) {
   }
 
   diagramElement.properties.push({
+    name: 'extension',
+    type: 'Extension'
+  },
+  {
     name: 'id',
     isAttr: true,
     isId: true,
@@ -94,6 +98,17 @@ module.exports = async function(results) {
   // filter DI
   model.types = model.types.filter(({ name }) => {
     return name && name.includes('di:');
+  });
+
+  model.types.push({
+    name: 'Extension',
+    properties: [
+      {
+        name: 'values',
+        isMany: true,
+        type: 'Element'
+      }
+    ]
   });
 
   model.enumerations = model.enumerations.filter(({ name }) => {

--- a/test/fixtures/dmn/dmndi/extension.dmn
+++ b/test/fixtures/dmn/dmndi/extension.dmn
@@ -1,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<definitions xmlns="https://www.omg.org/spec/DMN/20191111/MODEL/" xmlns:dmndi="https://www.omg.org/spec/DMN/20191111/DMNDI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" xmlns:custom="http://example.com" id="Definitions_1" name="DRD" namespace="http://camunda.org/schema/1.0/dmn">
+  <decision id="Decision_1" name="Decision 1" />
+  <dmndi:DMNDI>
+    <dmndi:DMNDiagram>
+      <di:extension>
+        <custom:Element id="CustomElement" />
+      </di:extension>
+      <dmndi:DMNShape dmnElementRef="Decision_1">
+        <dc:Bounds height="80" width="180" x="100" y="100" />
+      </dmndi:DMNShape>
+    </dmndi:DMNDiagram>
+  </dmndi:DMNDI>
+</definitions>

--- a/test/spec/xml/read.js
+++ b/test/spec/xml/read.js
@@ -590,6 +590,26 @@ describe('dmn-moddle - read', function() {
       });
     });
 
+
+    it('dmndi:Diagram with di:Extension', async function() {
+
+      // when
+      const definitions = await read('test/fixtures/dmn/dmndi/extension.dmn');
+
+      // then
+      const diagram = definitions.get('dmnDI').diagrams[0];
+
+      expect(diagram.get('extension')).to.jsonEqual({
+        $type: 'di:Extension',
+        values: [
+          {
+            $type: 'custom:Element',
+            id: 'CustomElement'
+          }
+        ]
+      });
+    });
+
   });
 
 

--- a/test/spec/xml/roundtrip.js
+++ b/test/spec/xml/roundtrip.js
@@ -46,6 +46,9 @@ describe('dmn-moddle - roundtrip', function() {
 
     it('with Size', roundtrip('test/fixtures/dmn/dmndi/size.dmn'));
 
+
+    it('with di:Extension', roundtrip('test/fixtures/dmn/dmndi/extension.dmn'));
+
   });
 
 

--- a/test/spec/xml/write.js
+++ b/test/spec/xml/write.js
@@ -341,6 +341,50 @@ describe('dmn-moddle - write', function() {
       expect(xml).to.equal(expected);
     });
 
+
+    it('di:Extension', async function() {
+
+
+      // given
+      const expected =
+        '<dmn:definitions xmlns:dmn="https://www.omg.org/spec/DMN/20191111/MODEL/" xmlns:dmndi="https://www.omg.org/spec/DMN/20191111/DMNDI/" xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/">' +
+          '<dmndi:DMNDI>' +
+            '<dmndi:DMNDiagram id="DMNDiagram_1">' +
+              '<di:extension />' +
+            '</dmndi:DMNDiagram>' +
+          '</dmndi:DMNDI>' +
+        '</dmn:definitions>';
+
+      const definitions = moddle.create('dmn:Definitions');
+
+      const dmnDiagram = moddle.create('dmndi:DMNDiagram', {
+        id: 'DMNDiagram_1',
+        diagramElements: []
+      });
+
+      const dmnDI = moddle.create('dmndi:DMNDI', {
+        diagrams: [ dmnDiagram ]
+      });
+
+      dmnDiagram.$parent = dmnDI;
+
+      definitions.set('dmnDI', dmnDI);
+
+      dmnDI.$parent = definitions;
+
+      const extension = moddle.create('di:Extension');
+
+      dmnDiagram.set('extension', extension);
+
+      extension.$parent = dmnDiagram;
+
+      // when
+      const xml = await write(definitions);
+
+      // then
+      expect(xml).to.equal(expected);
+    });
+
   });
 
 

